src/router_sc_worker.erl

```diff
@@ -298,18 +298,25 @@ open_next_state_channel(#state{pubkey = PubKey, sig_fun = SigFun, oui = OUI, cha
     Ledger = blockchain:ledger(Chain),
     {ok, ChainHeight} = blockchain:height(Chain),
     NextExpiration =
-        case active_sc_expiration() of
-            {error, no_active_sc} ->
+        case sc_expiration() of
+            {error, _Reason} ->
+                lager:info("failed to get a good expiration ~p", [_Reason]),
                 %% Just set it to expiration_interval
                 get_sc_expiration_interval();
             {ok, ActiveSCExpiration} ->
                 %% We set the next SC expiration to the difference between
                 %% current chain height and active plus the expiration_interval
-                abs(ActiveSCExpiration - ChainHeight) + get_sc_expiration_interval()
+                Max = blockchain_utils:approx_blocks_in_week(Ledger),
+                Expiration = abs(ActiveSCExpiration - ChainHeight) + get_sc_expiration_interval(),
+                case Expiration > Max of
+                    false ->
+                        Expiration;
+                    true ->
+                        Expiration - (Expiration - Max)
```

Comment:
Is this `min(Expiration, Max)`? 

Do we want to log if we're changing an expected the expiration, so we can know when we're overreaching?